wapi/session.py

```diff
@@ -478,11 +489,20 @@ def data_request(self, req_type, urlbase, url, data=None, rawdata=None, authval=
         if (timeout is not None or (500 <= res.status_code < 600) or res.status_code == 408) and retries > 0:
             if RETRY_DELAY > 0:
                 time.sleep(RETRY_DELAY)
-            return self.data_request(req_type, urlbase, url, data, rawdata, authval, stream, retries-1)
+            return self.send_data_request(req_type, urlbase, url, data, rawdata, authval, stream, retries-1)
```

Comment:
the arguments given dont match the expected ones in the signature - you forgot headers-val between rawdata and authval

Reply:
Nice catch. I have added one test for that as well.